sklearn/cluster/_birch.py

```diff
@@ -92,7 +92,7 @@ def _split_node(node, threshold, branching_factor):
 
     node1_closer = node1_dist < node2_dist
     for idx, subcluster in enumerate(node.subclusters_):
-        if node1_closer[idx]:
```

Comment:
Maybe add a comment explaining the edge case? I remember trying to understand last week how this was related to duplicates and not really managing to ... full disclosure, I am not super familiar with BIRCH.

Reply:
I moved the idx check directly close to where we define which subcluster is closest to which node so that I could add a clear comment on what's happening.

What happens is:
- you want to split a node. The node has several centroids.
- for that you compute all distances bewteen all centroids. The new nodes will be defined from the 2 centroids that are the most far apart.
- if it happens that the node only contains duplicates, then all centroids are the same point and all distances will be zero. Then the mask `node1_closer = node1_dist < node2_dist` considers that the centroid defining node1 is closer to the one defining node2 than to itself.

Reply:
OK, great thanks a lot for the explanation and the additional comment, merging!